x-pack/plugins/security_solution/public/timelines/store/timeline/epic_changed.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Action } from 'redux';
+import type { Epic } from 'redux-observable';
+import { get } from 'lodash/fp';
+import { filter, map } from 'rxjs/operators';
+
+import {
+  applyKqlFilterQuery,
+  addProvider,
+  dataProviderEdited,
+  removeColumn,
+  removeProvider,
+  updateColumns,
+  updateEqlOptions,
+  updateDataProviderEnabled,
+  updateDataProviderExcluded,
+  updateDataProviderType,
+  updateKqlMode,
+  updateProviders,
+  updateRange,
+  updateSort,
+  upsertColumn,
+  updateDataView,
+  updateTitleAndDescription,
+  setExcludedRowRendererIds,
+  setFilters,
+  setSavedQueryId,
+  setChanged,
+} from './actions';
+
+/**
+ * All action types that will mark a timeline as changed
+ */
+const timelineChangedTypes = new Set([
+  applyKqlFilterQuery.type,
+  addProvider.type,
+  dataProviderEdited.type,
+  removeProvider.type,
+  setExcludedRowRendererIds.type,
+  setFilters.type,
+  setSavedQueryId.type,
+  updateDataProviderEnabled.type,
+  updateDataProviderExcluded.type,
+  updateDataProviderType.type,
+  updateEqlOptions.type,
+  updateKqlMode.type,
+  updateProviders.type,
+  updateTitleAndDescription.type,
+
```

Comment:
@logeekal should we add the logic for es|ql here as well?

Reply:
@michaelolo24 , do you mean when `savedSearchId` is updated? I would agree that we can add it here..

Currently, @janmonschke  has put in `saveTimeline` logic in the ESQL/Discover update saved search module itself.

What do you think @janmonschke ?